src/libs/ReportListUtils.js

```diff
@@ -30,63 +47,149 @@ function getParticipantLogins(report) {
 }
 
 /**
- * Does the report have multiple participants?
+ * Returns the personal details for an array of logins
  *
- * @param {Object} report
- * @returns {Boolean}
+ * @param {Array} logins
+ * @returns {Array}
  */
-function reportHasMultipleParticipants(report) {
-    return getParticipantLogins(report).length > 1;
+function getPersonalDetailsForLogins(logins) {
+    return _.map(logins, (login) => {
+        let personalDetail = personalDetails[login];
+
+        if (!personalDetail) {
+            personalDetail = {
+                login,
+                displayName: login,
+                avatarURL: getDefaultAvatar(login),
+            };
+        }
+
+        return personalDetail;
+    });
 }
 
 /**
- * Is the login provided a report participant?
+ * Returns a string with all relevant search terms
  *
  * @param {Object} report
- * @param {String} login
- * @returns {Boolean}
+ * @param {Array} personalDetailList
+ * @return {String}
  */
-function isUserReportParticipant(report, login) {
-    return _.some(getParticipantLogins(report), participantLogin => participantLogin === login);
-}
+function getSearchText(report, personalDetailList) {
+    const searchTerms = [];
 
-/**
- * Check if the report has a draft comment
- *
- * @param {Object} comments
- * @param {Number} reportID
- * @returns {Boolean}
- */
-function hasComment(comments, reportID) {
-    const allComments = lodashGet(comments, `${ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT}${reportID}`, '');
-    return allComments.length > 0;
+    _.each(personalDetailList, (personalDetail) => {
+        searchTerms.push(personalDetail.displayName);
+        searchTerms.push(personalDetail.login);
+    });
+
+    if (report) {
+        searchTerms.push(...report.reportName.split(',').map(name => name.trim()));
+        searchTerms.push(...report.participants);
+    }
+
+    return _.unique(searchTerms).join(' ');
 }
 
 /**
  * Creates a report list option
  *
  * @param {Array<Object>} personalDetailList
  * @param {Object} [report]
- * @param {Object} [comments]
- * @param {Number} [activeReportID]
  * @returns {Object}
  */
-function createOption(personalDetailList, report, comments, activeReportID) {
-    // There is only one participant so return the single participant view
+function createOption(personalDetailList, report) {
```

Comment:
For the former section i've left a related comment [here](https://github.com/Expensify/Expensify.cash/pull/1252#discussion_r559859764).

Reply:
Thanks for the feedback. What do you think about maybe splitting this into two methods instead? 

Something like `createReportOption()` and `createPersonalDetailOption()` maybe this would make it easier to parse?

I guess I'm not really sure exactly what you meant by:

>we should have a single object param here called reportOrPersonalDetail and then use the appropriate keys to determine the option values

Which keys are you referring to?

As for null coalesce. I think the `||` operator feels like the same thing. I don't find it more or less readable personally, but I'm also not too sure about when it would be used or if we can use it yet (probably for sure with a babel plugin). There's an MDN doc on it [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator). Maybe something to bring up in `#engineering-chat`?

Reply:
> As for null coalesce. I think the || operator feels like the same thing. I don't find it more or less readable personally
Hmm maybe I'm in the minority but i find `||` a little unsettling to use because I natively use it for boolean. So having it return something other than boolean feels odd to me 🤷  whereas `??` is a clear non-comparison operator. But anyway, not a concern for the moment.